scripts/g.extension/g.extension.py

```diff
@@ -685,9 +687,19 @@ def install_extension(source, url, xmlurl):
     else:
         # for now it is reasonable to assume that only official source
         # will provide the metadata file
-        if source == 'official':
+        if source == 'official' and len(installed_modules) <= len(mlist):
             grass.message(_("Updating addons metadata file..."))
             blist = install_extension_xml(xmlurl, mlist)
+        if source == 'official' and len(installed_modules) > len(mlist):
+            grass.message(_("Updating addons metadata file..."))
+            blist = install_inofficial_extension_xml(tmp_dir, installed_modules)
+        else:
+            grass.message(_("Updating inofficial addons metadata file..."))
```

Comment:
```suggestion
            grass.message(_("Updating private addons metadata file..."))
```

Reply:
I am not sure which wording is better and more clear to the casual user..., you decide!

Reply:
A suggestion:
- **Official** are add-ons accepted officialy by the core development team
- **Unofficial** are add-ons still in a repository owned by the GRASS GIS team, yet not meeting all requirements to be official (such as: lacking unit tests and testing overall, lacking clean documentation and references to original work, and similar).
- **Private** would mean add-ons that "I" can only see in my local computing environment.